### PR TITLE
Documentation additions

### PR DESCRIPTION
**Pull request by danforbes**
Going to do this piecemeal because it will get out of control quickly.

### CODE REVIEW

File: crates/ggml/src/lib.rs [hunk 1]
```diff
@@ -29,11 +29,11 @@ pub type ElementType = Type;
 #[derive(Debug, PartialEq, Clone, Copy)]
 /// The format of the file containing the model.
 pub enum ContainerType {
-    /// `GGML`: legacy format, oldest ggml tensor file format
+    /// `Legacy format, oldest ggml tensor file format
```
Comment (philpax) [inline crates/ggml/src/lib.rs:32]: Accidental backtick left here

File: crates/llm-base/src/lib.rs [hunk 2]
```diff
@@ -32,18 +32,17 @@ pub use vocabulary::{TokenBias, TokenId, Vocabulary};
 pub struct InferenceParameters {
     /// The number of threads to use.
     pub n_threads: usize,
-    /// [InferenceSession::feed_prompt] processes the prompt in batches of tokens.
-    /// This controls how large an individual batch is.
+    /// Controls batch size for [InferenceSession::feed_prompt].
```
Comment (philpax) [inline crates/llm-base/src/lib.rs:35]: Explain what a batch is here

File: crates/llm/src/lib.rs [hunk 3]
```diff
@@ -1,8 +1,61 @@
...
-//! Large Language Models (LLMs) such as LLaMA.
+//! This crate provides a unified interface for loading and using
+//! Large Language Models (LLMs). The following models are supported:
 //!
-//! At present, the only supported backend is GGML, but this is expected to
+//! - [BLOOM](https://huggingface.co/docs/transformers/model_doc/bloom) (inference produces garbled results)
+//! - [GPT-2](https://huggingface.co/docs/transformers/model_doc/gpt2)
+//! - [GPT-J](https://huggingface.co/docs/transformers/model_doc/gptj)
+//! - [LLaMA](https://huggingface.co/docs/transformers/model_doc/llama)
+//! - [GPT-NeoX](https://huggingface.co/docs/transformers/model_doc/gpt_neox)
```
Comment (philpax) [inline crates/llm/src/lib.rs:8]: I'd suggest linking to the crates here (e.g. https://crates.io/crates/llm-neox), as you already link to HF in each crate's doc

**Review by philpax — APPROVED** [review]

**Comment by philpax** [comment]
LGTM, apart from the above 🚀 ! I'd suggest adding `#![deny(missing_docs)]` to each of the crates if not already present, so that the compiler will tell you what needs to be documented.

**Comment by danforbes** [comment]
I've addressed all your comments (I hope) and added a few extra docs...I'd love some help filling in the docs for the hyperparamaters that I wasn't able to do myself. Please feel free to make any changes you'd like and merge this when you see fit. I will continue working on docs in separate PRs to keep things manageable.

**Comment by philpax** [comment]
Looks good to me. Yeah, the hyperparameters are... not the most descriptive, I didn't document them because I wasn't sure myself. I'll have to get back to that at some point.

I'll merge this, and we can keep iterating on it. Great work so far!

**Pull request merged**